### DISCUSSION TITLE
Preferred way to add new redfish devices

### DISCUSSION THREAD

**Discussion by mat-ti** [bb-Ricardo/netbox-sync#410] [Q&A]
Hi,
I'm using the check_redfish script to generate the json files and want to add them to netbox with the netbox-sync script. Maybe I'm missing something small detail, these completly fresh devices can't be added with [source/redfish]:
`ERROR: Unable to find device with id 'None' or serial '<serial>' in NetBox inventory from inventory file`

Which way should I choose to add complete new devices to netbox?

**Comment by bb-Ricardo** [comment]
Hi,

That issue correct. Due to NetBox constraints it is not possible to add device without certain attributes. These are unknown to the script and therefore the device needs to exists in NetBox before you can use the script to update the attributes.

**Comment by mat-ti** [comment]
ok, got it. In which way should I add new devices to netbox? What solutions are a using others?
